pkg/protocol/engine/filter/blockfilter/filter.go

```diff
@@ -69,6 +76,29 @@ func (f *Filter) ProcessReceivedBlock(block *model.Block, source peer.ID) {
 		return
 	}
 
+	if _, isValidation := block.ValidationBlock(); isValidation {
+		blockAPI, err := f.apiProvider.APIForVersion(block.ProtocolBlock().ProtocolVersion)
+		if err != nil {
+			f.events.BlockPreFiltered.Trigger(&filter.BlockPreFilteredEvent{
+				Block:  block,
+				Reason: ierrors.Wrapf(err, "could not get API for version %d", block.ProtocolBlock().ProtocolVersion),
+				Source: source,
+			})
+
+			return
+		}
+		blockSlot := blockAPI.TimeProvider().SlotFromTime(block.ProtocolBlock().IssuingTime)
```

Comment:
Wait, I thought that candidate validators should always issue validation blocks as well, so we can't issue them here. Or am I missing something?

Reply:
non-committee validators must issue basic blocks using their own Mana in order to put themselves forward for committee selection. Validation blocks are now reserved only for committee.